tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -332,7 +374,151 @@ TEST_F(OverlapTest, ReduceScatterBasedPipeliningHostIrImplementation) {
 
     hie.runWithInput(std::move(inputs));
   }
-  validate();
 }
 
+using RingBasedOverlapTestParams = std::tuple<
+    bool, // use_coalescence
```

Comment:
Thanks for the nice explanation in the PR description. It would be really great if the comment stays with the code. It's much harder for readers to `git blame`  a PR that made a particular change.

Reply:
Sure, let me add a comment

Reply:
Thinking about it, I think it would make more sense to add a detailed comment where we add the HostIr for coalesced group, in a PR that I will open soon. Does that sound good to you ?

Reply:
I'm unsure how relevant this is, because you may want to remove this knob for now. Anyhow, I copied some of what you said in the description in another comment. If you decide to keep the knob, I'd keep the comment with the code.

Reply:
I removed the knob